packages/libs/eda/src/lib/map/analysis/MarkerConfiguration/BarPlotMarkerConfigurationMenu.tsx

```diff
@@ -26,16 +38,105 @@ interface Props
   > {
   onChange: (configuration: BarPlotMarkerConfiguration) => void;
   configuration: BarPlotMarkerConfiguration;
+  overlayConfiguration: OverlayConfig | undefined;
+  overlayVariable: Variable | undefined;
+  subsettingClient: SubsettingClient;
+  studyId: string;
+  filters: Filter[] | undefined;
+  continuousMarkerPreview: JSX.Element | undefined;
+  /**
+   * Always used for categorical marker preview. Also used in categorical table if selectedCountsOption is 'filtered'
+   */
+  allFilteredCategoricalValues: AllValuesDefinition[] | undefined;
+  /**
+   * Only defined and used in categorical table if selectedCountsOption is 'visible'
+   */
+  allVisibleCategoricalValues: AllValuesDefinition[] | undefined;
 }
 
+// TODO: generalize this and PieMarkerConfigMenu into MarkerConfigurationMenu. Lots of code repetition...
+
 export function BarPlotMarkerConfigurationMenu({
   entities,
   onChange,
   starredVariables,
   toggleStarredVariable,
   configuration,
   constraints,
+  overlayConfiguration,
+  overlayVariable,
+  subsettingClient,
+  studyId,
+  filters,
+  continuousMarkerPreview,
+  allFilteredCategoricalValues,
+  allVisibleCategoricalValues,
 }: Props) {
+  /**
+   * Used to track the CategoricalMarkerConfigurationTable's selection state, which allows users to
+   * select more than the allowable limit. Doing so results in a message to the user that they've selected
+   * too many values. The state is lifted up (versus living in CategoricalMarkerConfigurationTable) in order
+   * to pass its length to CategoricalMarkerPreview.
+   */
+  const [uncontrolledSelections, setUncontrolledSelections] = useState(
+    new Set(
+      overlayConfiguration?.overlayType === 'categorical'
+        ? overlayConfiguration?.overlayValues
+        : undefined
+    )
+  );
+  useEffect(() => {
+    if (overlayConfiguration?.overlayType !== 'categorical') return;
+    setUncontrolledSelections(new Set(overlayConfiguration?.overlayValues));
+  }, [overlayConfiguration?.overlayValues, overlayConfiguration?.overlayType]);
```

Comment:
You can remove the `useEffect` here. See https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes.

```suggestion
  if (overlayConfiguration?.overlayType === 'categorical') {
    setUncontrolledSelections(new Set(overlayConfiguration?.overlayValues));
  }
```

Reply:
This suggestion results in an infinite loop, which is why I think I reached for the `useEffect`. But it does appear that the "Better" suggestion from that section of the React docs works. Pretty clunky looking though!
```js
  const [previousOverlays, setPreviousOverlays] = useState(overlayConfiguration?.overlayValues);
  if (
      previousOverlays !== overlayConfiguration?.overlayValues && 
      overlayConfiguration?.overlayType === 'categorical'
    ) {
    setUncontrolledSelections(new Set(overlayConfiguration?.overlayValues));
    setPreviousOverlays(overlayConfiguration.overlayValues);
  }
```
Thoughts on this?

Reply:
Ah, yes, I see. This is more verbose, but also more efficient. Since this logic will be used in at least two places, maybe extract it into a hook. You can also add the initialization of `uncontrolledSelections` in the hook and call it `useUncontrolledSelections`. That will make it easier for the reader to ignore the implementation details, if they want to.